src/Keboola/S3Extractor/Extractor.php

```diff
@@ -282,4 +273,54 @@ public function extract($outputPath)
             return [];
         }
     }
+
+    private function loginViaCredentials(): S3Client
+    {
+        $awsCred = new Credentials($this->config->getAccessKeyId(), $this->config->getSecretAccessKey());
+        return new S3Client([
+            'region' => $this->getBucketRegion($awsCred),
+            'version' => '2006-03-01',
+            'credentials' => $awsCred,
+        ]);
+    }
+
+    private function loginViaRole(): S3Client
+    {
+        $awsCred = new Credentials(getenv('KEBOOLA_USER_AWS_ACCESS_KEY'), getenv('KEBOOLA_USER_AWS_SECRET_KEY'));
```

Comment:
Chapem to spravne, ze tieto premenne prostredia definuje KBC? Ako to bude fungovat? Urobi sa to nejak specialne pre tuto komponentu? ... cakal by som, ze sa to vygeneruje v UI a ulozi do configu (z toho co som si precital v issue), ale asi tomu nechapem spravne, ... vysvetlis mi to prosim v skratke? aby som to mohol skontrolovat, .... vdaka ;)

Reply:
díky, na to jsem zapomněl zjistit

no měl by u nás v produkčním aws být vytvořený jeden uživatel který bude mít jen oprávnění na assumeRole a ten bude přepínat na konkrétní roli - takže by to konfigurací jít asi nemělo, spíš nastavit envy do produkce aby to nikde nebylo vidět, diskuze okolo bude na slacku:
https://keboola.slack.com/archives/CQ1ASK06M/p1590653013011700